### PR TITLE
Transactional rename on Windows

### PR DESCRIPTION
**Pull request by mehrdadn**

### CODE REVIEW

File: client/sources/ok_test/models.py [hunk 1]
```diff
@@ -179,15 +180,22 @@ def dump(self):
...
-        else:
-            # Use an atomic rename operation to prevent test corruption
-            os.replace(test_tmp, self.file)
+        success = False
+        try:
+            storage.replace_transactional(test_tmp, self.file)
+            success = True
+        except (NotImplementedError, OSError):
+            pass
+        if not success:
```
Comment (colinschoen) [inline client/sources/ok_test/models.py]: Would it make sense for this block to be inside the body of the `except` clause and the `success = False; success = True` bit go away.

Comment (colinschoen) [reply client/sources/ok_test/models.py]: `ok-client` should be python3 only, but I'm curious about that issue. There are issues raising different exceptions in an except block in python2?

Comment (mehrdadn) [reply client/sources/ok_test/models.py]: Yup! What would Python print? :-)
```
try:
	try: raise ValueError("ValueError")
	except:
		try: raise NotImplementedError("NotImplementedError")
		except: pass
		raise
except Exception as ex: print(ex)
```

Comment (mehrdadn) [reply client/sources/ok_test/models.py]: Okay I just removed `success`, see the updated code.

Comment (mehrdadn) [reply client/sources/ok_test/models.py]: Oh I just realized I don't raise `NotImplementedError` properly... let me fix that.

Comment (mehrdadn) [reply client/sources/ok_test/models.py]: Ok fixed.

**Pull request merged**